packages/frames.js/src/core/middleware/openframes.ts

```diff
@@ -0,0 +1,107 @@
+import { ClientProtocolId } from "../..";
+import { RequestBodyNotJSONError } from "../errors";
+import { FramesMiddleware, FramesMiddlewareReturnType } from "../types";
+import { isFrameDefinition } from "../utils";
+
+type FramesMessageContext = {
+  // TODO: this should extend a base FrameMessage with common fields
+  message?: any;
```

Comment:
what happens if you replace any with some object like `{ test: string}`? Does it merge the declaration so it is available on `message` in `ctx` ?

Reply:
@michalkvasnicak yes it does

Reply:
so that should solve the issue right?

Reply:
ah I see... I think the only way to narrow down the type is to use discriminated unions. Or provide the message under different property name and then you can decide, soemthing like `ctx.openFramesMessage`.